src/ray/raylet/node_manager.cc

```diff
@@ -810,13 +810,13 @@ void NodeManager::HandleActorStateTransition(const ActorID &actor_id,
     } else {
       // Only process the state transition if it is to a later state than ours.
       if (actor_registration.GetState() > it->second.GetState() &&
-          actor_registration.GetRemainingReconstructions() ==
-              it->second.GetRemainingReconstructions()) {
+          actor_registration.GetNumRestarts() == it->second.GetNumRestarts()) {
         // The new state is later than ours if it is about the same lifetime, but
         // a greater state.
         it->second = actor_registration;
-      } else if (actor_registration.GetRemainingReconstructions() <
-                 it->second.GetRemainingReconstructions()) {
+        // imagine remaining reconst 4 max 10 num reconst 6
+        // other remaining reconst 4 max 10 num reconst 6
```

Comment:
Not sure if these comments are intentionally left here, but if so please make them more clear